application/apps/rustcore/ts-bindings/spec/session.benchmark.spec.ts

```diff
@@ -0,0 +1,306 @@
+// tslint:disable
+
+// We need to provide path to TypeScript types definitions
+/// <reference path="../node_modules/@types/jasmine/index.d.ts" />
+/// <reference path="../node_modules/@types/node/index.d.ts" />
+import { initLogger } from './logger';
+initLogger();
+import { Session, Factory } from '../src/api/session';
+import { IAttachmentsUpdatedUpdated } from '../src/api/session.provider';
+import { IAttachment } from 'platform/types/content';
+import { createSampleFile, finish, performanceReport, setMeasurement, runner } from './common';
+import { readBenchmarkConfigurationFile } from './config_benchmarks';
+import { IndexingMode } from 'platform/types/content';
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+
+const config = readBenchmarkConfigurationFile().get().tests.benchmark;
+
+describe('Benchmark Tests', function () {
+    Object.keys(config.performance.tests).forEach((alias: string) => {
+        const test = config.performance.tests[alias];
+        const testName = `${test.alias}`;
+        if (test.ignore) {
+            console.log(`Test "${testName}" has been ignored`);
+            return;
+        }
+
+        it(testName, function () {
+            return runner(
+                    {
+                        open_as: '',
+                        ignore: false,
+                        alias: testName,
+                        expectation_ms: 10000,
+                        file: '',
+                    },
+                    1,
+                    async (logger, done, collector) => {
+                    const measurement = setMeasurement();
+                    try {
+                        const session = await Session.create();
+                        session.debug(true, testName);
+
+                        const stream = session.getStream();
+                        if (stream instanceof Error) throw stream;
+
+                        const events = session.getEvents();
+                        if (events instanceof Error) throw events;
+
+                        const search = session.getSearch();
```

Comment:
In general, this block of code 
```
                        const session = await Session.create();
                        session.debug(true, testName);

                        const stream = session.getStream();
                        if (stream instanceof Error) throw stream;

                        const events = session.getEvents();
                        if (events instanceof Error) throw events;

                        const search = session.getSearch();
                        if (search instanceof Error) throw events;
```
Can be extracted into standalone function. Maybe makes sense also update/refactor others tests.
```
async function getTestSessionEnv(testName: string): Promise<{
    session: Session;
    events: ISessionEvents;
    stream: SessionStream;
    search: SessionSearch;
}> {
    const session = await Session.create();
    session.debug(true, testName);

    const stream = session.getStream();
    if (stream instanceof Error) {
        await session.destroy();
        return Promise.reject(stream);
    }

    const events = session.getEvents();
    if (events instanceof Error) {
        await session.destroy();
        return Promise.reject(events);
    }

    const search = session.getSearch();
    if (search instanceof Error) {
        await session.destroy();
        return Promise.reject(search);
    }

    return { session, events, search, stream };
}

```
This function can be called each time as soon as you need session and its dependencies (events, search, stream)

Probably make sense to create this function in `common` to reused it not only in benchmarks tests, but in regulars also